encoder.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import random as r
+
+from tensorflow.keras.models import load_model
+from tensorflow.keras.applications.vgg16 import VGG16, preprocess_input
+from PIL import Image
+from matplotlib.pyplot import imshow
+import dlib
+import scipy
+
+import tensorflow as tf
+from tensorflow.keras.models import Model
+from tensorflow.keras.losses import mean_squared_error,logcosh
+from tensorflow.keras.optimizers import Adam
+
+from PIL import Image
+from matplotlib.pyplot import imshow
+from tqdm import tqdm
+from time import time
+
+from utils.utils_stylegan2 import convert_images_to_uint8
+from stylegan2_generator import StyleGan2Generator
+
+from sklearn.model_selection import train_test_split
+from scipy.spatial import distance
+import matplotlib.pylab as plt
+import json
+
+import os
+import subprocess
+import re
+import sys
+import imageio
+import argparse
+
+# In[7]:
+
+
+parser = argparse.ArgumentParser(description='Convert photo to GIF.')
+parser.add_argument('filename', metavar='filename', type=str,
+                   help='a filename to process (in JPG, but without the extension)')
+parser.add_argument('--gpu', dest='use_gpu', action='store_const',
+                   const=True, default=False,
+                   help='Whether to use GPU or not(CUDA enabled machines)')
+
+args = parser.parse_args()
+filename = args.filename
+impl = "ref" if args.use_gpu is False else "cuda"
+
+# In[14]:
+
+
+weights = "ffhq"
+
+
+# In[15]:
+
+
+detector = dlib.get_frontal_face_detector()
+shape_predictor = dlib.shape_predictor("weights/shape_predictor_68_face_landmarks.dat")
+vgg16=VGG16(include_top=False,input_shape=(224,224,3),weights='imagenet')
+vggmod=Model(vgg16.input, vgg16.layers[6].output)
+truncation_psi = 0.5
+w_average = np.load('weights/{}_dlatent_avg.npy'.format(weights))
+generator = StyleGan2Generator(weights="ffhq", impl=impl, gpu=args.use_gpu)
+
+
+# In[3]:
+
+
+class VGG(Model):
+    def __init__(self,model):
+        super(VGG,self).__init__(name = 'my_model')
+        self.gen = model
+    def call(self,inputs):
+        x = self.gen(inputs)
+        return x
+
+
+# In[4]:
+
+
+vgg = VGG(vggmod)
+vgg.trainable = False
+
+
+# In[5]:
```

Comment:
The definition of the initializer (in your case the imagenet model) should be in another class. E.g Initializer, where you import the model defined in other file.